docs/Modules/common-modules/event-bus.md

```diff
@@ -9,4 +9,185 @@ parent: "Common Modules"
+
+* [on](#on)
+* [once](#once)
+* [off](#off)
+* [createEvent](#createEvent)
+* [fire](#fire)
+* [handleError](#handleError)
+
+## on
+
+### Usage
+
+on(events: _String_ or _Array< string >_, [priority=1000]: _number_, callback: _Function_, [that]:_Object_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `priority`: optional parameter specifying the priority with which the listener is called, where a higher number represents a higher priority and therefore leads to the listener being called earlier. If not provided, default of `1000` will be used.
+* `callback`: function which will be invoked with `event, ...additionalArguments`. Returning anything but `undefined` from this will stop the listener propagation. Returning `false` will prevent the execution of the default action for that event.
+*  `that`: optional parameter which can be used to pass context (`this`) to the callback.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+eventBus.on('saveSVG.start', 1500, function(event) {
+
+  // access event type
+  event.type; // 'saveSVG.start'
+
+  // stop propagation to other listeners
+  event.stopPropagation();
+
+  // prevent event default
+  event.preventDefault();
+
+  // Some custom logic
+  // [...]
+});
+```
+
+
+## once
+
+### Usage
+
+once(events: _String_ or _Array< string >_, [priority=1000]: _number_, callback: _Function_, [that]:_Object_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `priority`: optional parameter specifying the priority with which the listener is called, where a higher number represents a higher priority and therefore leads to the listener being called earlier. If not provided, default of `1000` will be used.
+* `callback`: function which will be invoked with `event, ...additionalArguments`. Returning anything but `undefined` from this will stop the listener propagation. Returning `false` will prevent the execution of the default action for that event.
+*  `that`: optional parameter which can be used to pass context (`this`) to the callback.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+function listener(event) {
+
+  // Perform some operation you only want to do once
+  // [...]
+
+  return 'One-time operation finished';
+}
+
+eventBus.once('onceEvent', listener);
+
+const firstEventFired = eventBus.fire('onceEvent'),
+      secondEventFired = eventBus.fire('onceEvent');
+
+console.log(firstEventFired); // 'One-time operation finished'
+console.log(secondEventFired); // undefined
+```
+
+## off
+
+### Usage
+
+off(events: _String_ or _Array< string >_, [callback]: _Function_): _undefined_
+
+### Parameters
+
+* `events`: name of the event or _Array_ of event names.
+* `callback`: optional parameter specifying the callback listener to be removed.
+
+### Return Value
+
+_undefined_
+
+### Code Example
+
+```javascript
+function listener1(event) {
```

Comment:
I'd go for an even simple example that doesn't also show priorities and return values.

Something like this:

```javascript
function fooListener(event) {
  console.log('Foo');
}

function barListener(event) {
  console.log('Bar');
}

eventBus.on('foo', fooListener);
eventBus.on('foo', barListener);

eventBus.fire('foo'); // 'Foo Bar'

eventBus.off('foo', fooListener);

eventBus.fire('foo'); // 'Bar'

eventBus.off('foo');

eventBus.fire('foo'); // ''
```

Reply:
Good point